tests/gocase/unit/introspection/replication_test.go

```diff
@@ -0,0 +1,349 @@
+
+	t.Run("resume broken transfer based files", func(t *testing.T) {
+		// Try to transfer some files, because max-replication-mb 1,
+		// so maybe more than 5 files are transferred for sleep 5s.
+		util.SlaveOf(t, slaveClient, master)
+		time.Sleep(5 * time.Second)
+
+		// Restart master server, let the slave try to full sync with master again,
+		// because slave already received some SST files, so we will skip them.
+		master.Restart()
+		masterClient.Close()
+		masterClient = master.NewClient()
+
+		require.NoError(t, masterClient.ConfigSet(ctx, "max-replication-mb", "0").Err())
+		require.Eventually(t, func() bool {
+			return slave.LogFileMatches(t, ".*skip count: 1.*")
+		}, 50*time.Second, 1000*time.Millisecond)
+		util.WaitForSync(t, slaveClient)
+		require.Equal(t, "b", slaveClient.Get(ctx, "a").Val())
+	})
+}
+
+func TestReplicationShareCheckpoint(t *testing.T) {
+	master := util.StartServer(t, map[string]string{})
+	defer master.Close()
+	masterClient := master.NewClient()
+	defer func() { require.NoError(t, masterClient.Close()) }()
+	util.Populate(t, masterClient, "", 1024, 1)
+	ctx := context.TODO()
+	require.NoError(t, masterClient.Set(ctx, "a", "b", 0).Err())
+	require.NoError(t, masterClient.Do(ctx, "compact").Err())
+	time.Sleep(time.Second)
+
+	slave1 := util.StartServer(t, map[string]string{})
+	defer slave1.Close()
+	slave1Client := slave1.NewClient()
+	defer func() { require.NoError(t, slave1Client.Close()) }()
+	util.Populate(t, slave1Client, "", 1026, 1)
+
+	slave2 := util.StartServer(t, map[string]string{})
+	defer slave2.Close()
+	slave2Client := slave2.NewClient()
+	defer func() { require.NoError(t, slave2Client.Close()) }()
+	util.Populate(t, slave2Client, "", 1026, 1)
+
+	t.Run("two slaves share one checkpoint for full replication", func(t *testing.T) {
+		util.SlaveOf(t, slave1Client, master)
+		util.SlaveOf(t, slave2Client, master)
+
+		require.Eventually(t, func() bool {
+			return master.LogFileMatches(t, ".*Use current existing checkpoint.*")
+		}, 50*time.Second, 100*time.Millisecond)
+		util.WaitForSync(t, slave1Client)
+		util.WaitForSync(t, slave2Client)
+		require.Equal(t, "b", slave1Client.Get(ctx, "a").Val())
+		require.Equal(t, "b", slave2Client.Get(ctx, "a").Val())
+	})
+}
+
+func TestReplicationContinueRunning(t *testing.T) {
+	master := util.StartServer(t, map[string]string{})
+	defer master.Close()
+	masterClient := master.NewClient()
+	defer func() { require.NoError(t, masterClient.Close()) }()
+
+	slave := util.StartServer(t, map[string]string{})
+	defer slave.Close()
+	slaveClient := slave.NewClient()
+	defer func() { require.NoError(t, slaveClient.Close()) }()
+
+	util.SlaveOf(t, slaveClient, master)
+	util.WaitForSync(t, slaveClient)
+
+	t.Run("Master doesn't pause replicating with replicas, #346", func(t *testing.T) {
+		ctx := context.TODO()
+		// In #346, we find a bug, if one command contains more than special
+		// number updates, master won't send replication stream to replicas.
```

Comment:
If it refers to Redis's issue 346, we may specify it clearly.

Reply:
Sure, it's just copied from previous test cases, we should claim it clearly.

Reply:
@tisonkun I took a look the issue and it's Kvrocks's issue, so it's ok to keep this description.